src/main/java/seedu/address/logic/parser/TimeslotRegex.java

```diff
@@ -0,0 +1,14 @@
+package seedu.address.logic.parser;
+
+/**
+ * Contains Regex Format for Date recognition in Raw User Inputs
+ */
+public class TimeslotRegex {
```

Comment:
i don't think there's a need to have a new class for this? i think it is fine to have these constants be in `TimeslotParser.java`

@Jacob-Pang for input

at most, i would probably make it an enum with a field, ie
```
enum TimeslotRegex {
    DATE_SLASH_SHORT ("([0-9]{2})/([0-9]{2})/([0-9]{2})"),
    DATE_SLASH_LONG ("([0-9]{2})/([0-9]{2})/([0-9]{4})"),
    DATE_DASH_SHORT ("([0-9]{2})-([0-9]{2})-([0-9]{2})"),
    DATE_DASH_LONG ("([0-9]{2})-([0-9]{2})-([0-9]{4})")

    private final String regex;

    public TimeslotRegex(String regex) {
        this.regex = regex;
    }

    public String getRegex() {
        return regex;
    }
}
```

you could then access the first regex by: `TimeslotRegex.DATE_SLASH_SHORT.getRegex()`

Reply:
Here is my take on this:
The enum could store the DateTimeFormatters instead,
```
enum StandardDateTimeFormatter {
    DATE_SLASH_SHORT ( "dd/MM/yyyy" )
    ...
    private dateTimePattern;
    private StandardDateTimeFormatter (String dateTimePattern) { 
        this.dateTimePattern = dateTimePattern; 
    }

    public DateTimeFormatter getDateTimeFormatter() {
        return DateTimeFormatter.ofPattern(dateTimePattern);
    }

    public String getDateTimePattern() {
        return dateTimePattern;
    }
}
```
Then, the parsing of the non-Next inputs could be performed like this:
```
public LocalDateTime parseDateTime (String input) {
    for (StandardDateTimeFormatter standardDateTimeFormatter : StandardDateTimeFormatter.values()) {
        try {
            return LocalDatetime.parse(input, standardDateTimeFormatter);
        } catch (DateTimeParseException e) {
            continue; // no need for regex checking because check is performed by LocalDateTime.parse
        }
    }
    throw ParseException(...); // does not conform to any of the standard inputs
```
I really do not recommend having date and time parsed separately, but for discussion sake, one can implement 2 enums, standardDateFormat and standardTimeFormat and the loop can be modified like this:
```
public LocalDateTime parseDateTime (String input) {
    for (StandardDateTimeFormat standardDateFormat : StandardDateFormat.values()) {
        for (StandardTimeFormat standardTimeFormat : StandardTimeFormat.values()) {
            try {
                DateTimeFormatter standardDateTimeFormatter = standardDateFormat.getDatePattern()
                        + standardTimeFormat.getTimePattern();
                return LocalDatetime.parse(input, standardDateTimeFormatter);
            } catch (DateTimeParseException e) {
                continue; // no need for regex checking because check is performed by LocalDateTime.parse
            }
    }
    throw ParseException(...); // does not conform to any of the standard inputs
```

Reply:
Ok! I've gone with both suggestions:
- Deleted Regex class to use an enum in `TimeslotParser` class
- Merged both date and time for the non-next commands but of course means there are twice as many enum variables now, but beats creating another enum class just for time.